src/execution/operator/persistent/physical_copy_to_file.cpp

```diff
@@ -244,6 +253,19 @@ class CopyToFunctionGlobalState : public GlobalSinkState {
 	idx_t global_offset = 0;
 };
 
+CopyToFunctionGlobalState::~CopyToFunctionGlobalState() {
+	if (!initialized || finalized || created_files.empty()) {
+		return;
+	}
+	// If we reach here, the query failed before Finalize was called
+	auto &fs = FileSystem::GetFileSystem(context);
+	for (auto &file : created_files) {
+		if (fs.FileExists(file)) {
+			fs.RemoveFile(file);
```

Comment:
Maybe it's better to call `TryRemoveFile` directly?
- `TryRemove` already handles the "delete if exists" semantics
- (minor) It could potentially save one IO operation, it reads to me only `DeleteRequest` is needed here: https://github.com/duckdb/duckdb-httpfs/blob/1087eb50dbc3fa2ed7aa258ab13628940a36b9b1/src/s3fs.cpp#L946-L958

Reply:
> Looking at your second point, it seems DeleteRequest() is S3 specific, no?

Yes. My point is: if a certain filesystem implement their own `TryRemove`, we could potentially benefit from less IO operations.